kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -198,14 +198,105 @@ class KRATOS_API(KRATOS_CORE) CalculateEmbeddedNodalVariableFromSkinProcess : pu
     ///@name Life Cycle
     ///@{
 
+    /**
+     * @brief Get the Default Settings object
+     * This method returns the default parameters for this proces.
+     * Note that it is required to be static since it is called during
+     * the construction of the object so no instantation exists yet.
+     * @return Parameters Default parameters json string
+     */
+    static Parameters GetDefaultSettings()
+    {
+        Parameters default_settings(R"(
+        {
+            "base_model_part_name": "",
+            "skin_model_part_name": "",
+            "skin_variable_name": "",
+            "embedded_nodal_variable_name": "",
+            "level_set_type": "",
+            "buffer_position": 0,
+            "aux_model_part_name": "IntersectedElementsModelPart",
+            "linear_solver_settings": {
+                "preconditioner_type": "amg",
+                "solver_type": "amgcl",
+                "smoother_type": "ilu0",
+                "krylov_type": "cg",
+                "max_iteration": 1000,
+                "verbosity": 0,
+                "tolerance": 1e-8,
+                "scaling": false,
+                "block_size": 1,
+                "use_block_matrices_if_possible": true
+            }
+        }
+        )");
+
+        return default_settings;
+    }
+
+    /**
+     * @brief Construct a new Calculate Embedded Nodal Variable From Skin Process object
+     * Constructor with model and json settings
+     * @param rModel Model container
+     * @param rSettings Settings json string
+     */
+    CalculateEmbeddedNodalVariableFromSkinProcess(
+        Model &rModel,
+        Parameters rSettings)
+        : CalculateEmbeddedNodalVariableFromSkinProcess(
+            rModel.GetModelPart(rSettings["base_model_part_name"].GetString()),
+            rModel.GetModelPart(rSettings["skin_model_part_name"].GetString()),
+            [] (Parameters x) -> Parameters {x.ValidateAndAssignDefaults(GetDefaultSettings()); return x;} (rSettings))
```

Comment:
just wondering why u used a lambda...but it looks correct to me

Reply:
To check the parameters at construction time. Doing it in this way allows to initialize the member variables with the "old-fashioned" constructor to keep them `const`.

Reply:
ok